src/System.Device.Gpio/System/Device/Gpio/AlternatePinMode.cs

```diff
@@ -0,0 +1,128 @@
+        /// Default Gpio mode
+        /// </summary>
+        public static AlternatePinMode Gpio = new AlternatePinMode("GPIO", 0);
+
+        /// <summary>
+        /// The mode is not known
+        /// </summary>
+        public static AlternatePinMode Unknown = new AlternatePinMode("Unknown", 0xffffffff);
+
+        /// <summary>
+        /// Alternate mode 0 for Raspberry Pi
+        /// </summary>
+        public static AlternatePinMode Alt0 = new AlternatePinMode("Alt0", 0b100);
+
+        /// <summary>
+        /// Alternate mode 1 for Raspberry Pi
+        /// </summary>
+        public static AlternatePinMode Alt1 = new AlternatePinMode("Alt1", 0b101);
+
+        /// <summary>
+        /// Alternate mode 2 for Raspberry Pi
+        /// </summary>
+        public static AlternatePinMode Alt2 = new AlternatePinMode("Alt2", 0b110);
+
+        /// <summary>
+        /// Alternate mode 3 for Raspberry Pi
+        /// </summary>
+        public static AlternatePinMode Alt3 = new AlternatePinMode("Alt3", 0b111);
+
+        /// <summary>
+        /// Alternate mode 4 for Raspberry Pi
+        /// </summary>
+        public static AlternatePinMode Alt4 = new AlternatePinMode("Alt4", 0b011);
+
+        /// <summary>
+        /// Alternate mode 5 for Raspberry Pi
+        /// </summary>
+        public static AlternatePinMode Alt5 = new AlternatePinMode("Alt5", 0b010);
+
+        /// <summary>
+        /// Creates a new mode.
+        /// </summary>
+        /// <param name="modeName">Common name of the mode</param>
+        /// <param name="modeValue">Internal value to identify this mode</param>
+        public AlternatePinMode(string modeName, uint modeValue)
+        {
+            ModeName = modeName;
+            ModeValue = modeValue;
+        }
+
+        /// <summary>
+        /// User readable name of the mode
+        /// </summary>
+        public string ModeName
+        {
+            get;
+        }
+
+        /// <summary>
+        /// Internal value used for this mode
+        /// </summary>
+        public uint ModeValue
+        {
+            get;
+        }
+
+        /// <summary>
+        /// Equality implementation
+        /// </summary>
+        public virtual bool Equals(AlternatePinMode? other)
+        {
+            if (ReferenceEquals(other, null))
+            {
+                return false;
+            }
+
+            return other.ModeValue == ModeValue && other.ModeName == ModeName;
+        }
+
+        /// <inheritdoc />
+        public override bool Equals(object? obj)
+        {
+            AlternatePinMode? other = obj as AlternatePinMode;
+            if (ReferenceEquals(other, null))
+            {
+                return false;
+            }
+
+            return other.ModeValue == ModeValue && other.ModeName == ModeName;
+        }
+
+        /// <inheritdoc />
+        public override int GetHashCode()
+        {
+            return ModeValue.GetHashCode() ^ ModeName.GetHashCode();
```

Comment:
nope, use this instead https://docs.microsoft.com/en-us/dotnet/api/system.hashcode.combine?view=dotnet-plat-ext-3.1

As is it is really easy to create collisions (even though it is right now meant to be used with user coming data, we have historically seen this happening eventually)

Reply:
Not sure how that would be any better. Collisions are to be expected for this method, that should not cause any harm (except a performance degradation). 
Besides that this would require an extra dependency. System.HashCode is not generally available.

Reply:
fine